cdc/sink/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -247,3 +255,46 @@ func (m *DDLSink) Close() {
 		}
 	}
 }
+
+// asyncExecAddIndexDDLIfTimeout executes ddl in async mode.
+func (m *DDLSink) asyncExecAddIndexDDLIfTimeout(ctx context.Context, ddl *model.DDLEvent) error {
+	done := make(chan error, 1)
+	// wait for 1 second at most
+	tick := time.NewTimer(1 * time.Second)
+	defer tick.Stop()
+	log.Info("async exec ddl start",
+		zap.String("changefeedID", m.id.String()),
+		zap.Uint64("commitTs", ddl.CommitTs),
+		zap.String("ddl", ddl.Query))
+	go func() {
+		if err := m.execDDLWithMaxRetries(ctx, ddl); err != nil {
+			log.Error("async exec ddl failed",
+				zap.String("changefeedID", m.id.String()),
+				zap.Uint64("commitTs", ddl.CommitTs),
+				zap.String("ddl", ddl.Query))
+			done <- err
+			return
+		}
+		log.Info("async exec ddl done",
+			zap.String("changefeedID", m.id.String()),
+			zap.Uint64("commitTs", ddl.CommitTs))
+		done <- nil
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("async ddl exits as canceled",
+				zap.String("changefeedID", m.id.String()),
+				zap.Uint64("commitTs", ddl.CommitTs),
+				zap.String("ddl", ddl.Query))
+		case err := <-done:
+			return err
+		case <-tick.C:
```

Comment:
Is it possible to report an error due to tcp connection timeout during each retry? If possible, and the ticker time limit is exceeded, then the downstream ddl is lost. 

In addition, if this is the expected behavior, please add a comment to explain this problem.